lib/pdf_fill/hash_converter.rb

```diff
@@ -62,35 +62,36 @@ def overflow?(key_data, value, from_array_overflow = false)
       limit.present? && value.size > limit
     end
 
-    def add_to_extras(key_data, v, i, top_level_key, overflow: true)
+    def add_to_extras(key_data, v, i, overflow: true, array_key_data: nil)
       return if v.blank? || key_data.nil?
       return if key_data[:question_num].blank? || key_data[:question_text].blank?
 
       i = nil if key_data[:skip_index]
       v = "$#{v}" if key_data[:dollar]
       v = v.extras_value if v.is_a?(PdfFill::FormValue)
+      array_question_text = array_key_data.try(:[], :question_text)
```

Comment:
This is the sole new piece of machinery in HashConverter. The existing code already Just Works to support list-and-loops... except we need to grab this overarching question header text.